@claude　ビルドエラーを修正して

[baseline-browser-mapping] The data in this module is over two months old.  To ensure accurate Baseline data, please update: `npm i baseline-browser-mapping@latest -D`
⚠ No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

   ▲ Next.js 16.0.4 (Turbopack)

   Creating an optimized production build ...
[baseline-browser-mapping] The data in this module is over two months old.  To ensure accurate Baseline data, please update: `npm i baseline-browser-mapping@latest -D`
 ✓ Compiled successfully in 4.6s
   Running TypeScript ...
Failed to compile.

./app/household/page.tsx:560:58
Type error: Property 'category' does not exist on type 'PieLabelRenderProps'.

  558 |                               cy="50%"
  559 |                               outerRadius={80}
> 560 |                               label={(entry) => `${entry.category} ${entry.percentage.toFixed(1)}%`}
      |                                                          ^
  561 |                             >
  562 |                               {expenseSummary.map((entry) => (
  563 |                                 <Cell key={entry.category} fill={CATEGORY_COLORS[entry.category]} />
Next.js build worker exited with code: 1 and signal: null
Error: Process completed with exit code 1.

--- a/app/household/page.tsx
+++ b/app/household/page.tsx
@@ -14,7 +14,7 @@ import {
   getCategorySummary,
   getMonthsInRange,
 } from '@/app/utils/household';
-import type { TransactionType, CategoryType } from '@/app/types/household';
+import type { TransactionType, CategoryType, CategorySummary } from '@/app/types/household';
 
 export default function HouseholdPage() {
   const { transactions, addTransaction, updateTransaction, deleteTransaction } = useTransactions();
@@ -557,7 +557,7 @@ export default function HouseholdPage() {
                               cx="50%"
                               cy="50%"
                               outerRadius={80}
-                              label={(entry) => `${entry.category} ${entry.percentage.toFixed(1)}%`}
+                              label={(entry: CategorySummary) => `${entry.category} ${entry.percentage.toFixed(1)}%`}
                             >
                               {expenseSummary.map((entry) => (
                                 <Cell key={entry.category} fill={CATEGORY_COLORS[entry.category]} />
@@ -607,7 +607,7 @@ export default function HouseholdPage() {
                               cx="50%"
                               cy="50%"
                               outerRadius={80}
-                              label={(entry) => `${entry.category} ${entry.percentage.toFixed(1)}%`}
+                              label={(entry: CategorySummary) => `${entry.category} ${entry.percentage.toFixed(1)}%`}
                             >
                               {incomeSummary.map((entry) => (
                                 <Cell key={entry.category} fill={CATEGORY_COLORS[entry.category]} />

--- a/app/household/page.tsx
+++ b/app/household/page.tsx
@@ -557,7 +557,10 @@ export default function HouseholdPage() {
                               cx="50%"
                               cy="50%"
                               outerRadius={80}
-                              label={(entry: CategorySummary) => `${entry.category} ${entry.percentage.toFixed(1)}%`}
+                              label={(entry) => {
+                                const data = entry as unknown as CategorySummary;
+                                return `${data.category} ${data.percentage.toFixed(1)}%`;
+                              }}
                             >
                               {expenseSummary.map((entry) => (
                                 <Cell key={entry.category} fill={CATEGORY_COLORS[entry.category]} />
@@ -607,7 +610,10 @@ export default function HouseholdPage() {
                               cx="50%"
                               cy="50%"
                               outerRadius={80}
-                              label={(entry: CategorySummary) => `${entry.category} ${entry.percentage.toFixed(1)}%`}
+                              label={(entry) => {
+                                const data = entry as unknown as CategorySummary;
+                                return `${data.category} ${data.percentage.toFixed(1)}%`;
+                              }}
                             >
                               {incomeSummary.map((entry) => (
                                 <Cell key={entry.category} fill={CATEGORY_COLORS[entry.category]} />

--- a/app/household/page.tsx
+++ b/app/household/page.tsx
@@ -566,7 +566,7 @@ export default function HouseholdPage() {
                                 <Cell key={entry.category} fill={CATEGORY_COLORS[entry.category]} />
                               ))}
                             </Pie>
-                            <Tooltip formatter={(value: number) => formatCurrency(value)} />
+                            <Tooltip formatter={(value) => value !== undefined ? formatCurrency(value as number) : ''} />
                           </PieChart>
                         </ResponsiveContainer>
                       </div>
@@ -619,7 +619,7 @@ export default function HouseholdPage() {
                                 <Cell key={entry.category} fill={CATEGORY_COLORS[entry.category]} />
                               ))}
                             </Pie>
-                            <Tooltip formatter={(value: number) => formatCurrency(value)} />
+                            <Tooltip formatter={(value) => value !== undefined ? formatCurrency(value as number) : ''} />
                           </PieChart>
                         </ResponsiveContainer>
                       </div>

